csi/controller_server.go

```diff
@@ -604,6 +649,12 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 		return nil, status.Error(codes.InvalidArgument, "Snapshot ID must be provided")
 	}
 
+	snapshot, _ := cs.apiClient.Snapshot.ById(req.SnapshotId)
+	if snapshot == nil {
+		logrus.Debugf("Snapshot %v does not exist", req.SnapshotId)
+		return &csi.DeleteSnapshotResponse{}, nil
+	}
+
```

Comment:
This is incorrect, have a look at the decode snapshotID function.
A csi snapshot refers to a longhorn backup.

Reply:
Check the ActionBackupDelete below and see if it returns a notFound error code, then you can evaluate the error below.

Reply:
Thanks for the hint! I think we could check the existence of backup volume name to achieve the same result. Fix the original commit(https://github.com/longhorn/longhorn-manager/pull/825/commits/1efd1ab258a7d6dd3692a85160ba883c26762599).